### ISSUE TITLE
Does not depend on ffi correctly

### ISSUE THREAD

**Issue by hroncok** [hroncok/rubygem-admesh#2]
When I do: `gem install radmesh`, I don't get the `ffi` gem, and I get an error instead:

```
$ gem install radmesh
Fetching: radmesh-0.98.2.gem (100%)
Successfully installed radmesh-0.98.2
Parsing documentation for radmesh-0.98.2
Installing ri documentation for radmesh-0.98.2
Done installing documentation for radmesh after 1 seconds
1 gem installed
$ ruby test.rb 
/usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- ffi (LoadError)
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /home/users/h/hroncmir/.gem/ruby/2.1.0/gems/radmesh-0.98.2/lib/radmesh/cadmesh.rb:2:in `<top (required)>'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /home/users/h/hroncmir/.gem/ruby/2.1.0/gems/radmesh-0.98.2/lib/radmesh/stl.rb:1:in `<top (required)>'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /home/users/h/hroncmir/.gem/ruby/2.1.0/gems/radmesh-0.98.2/lib/radmesh.rb:2:in `<top (required)>'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:135:in `require'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
    from /usr/lib64/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:144:in `require'
    from test.rb:1:in `<main>'
```

radmesh should depend on ffi.

**Comment by hroncok** [comment]
While the Gemfile clearly has `gem 'ffi', '~>1.9'` :(

@jirutka  Can you please help me understand this?

**Comment by jirutka** [comment]
Sorry for late response. The problem is simple, Gemfile is not used for gems. This file is read by [Bundler](https://bundler.io/) and it’s intended for applications. It’s fine to use it even in gem, but just for development/test dependencies, e.g. if you want to use groups or other fancy stuff provided by Bundler.  

Ruby gem (i.e. library) is described just by [*.gemspec](http://guides.rubygems.org/specification-reference/) file; there’s information about author, version, …, runtime (`add_runtime_dependency`) and development dependencies (`add_development_dependency`).

gemspec file is like setup.py, Gemfile is like requirements.txt (but _much_ more advanced) in the Python world.

tl;dr: Add `s.add_runtime_dependency 'ffi', '~> 1.9'` to your `radmesh.gemspec`.

**Comment by hroncok** [comment]
Awesome explanation, thanks.
